benchmarks/graphics_pipeline/main.cpp

```diff
@@ -701,49 +454,296 @@ void ProjApp::Setup()
+        lowPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * lowPlanarPositionBufferPtr->GetElementSize());
+        highPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * highPlanarPositionBufferPtr->GetElementSize());
+
+        // Iterate through the full vertex buffers, changing position data and appending indices.
+        //
+        // i : sphere index
+        // j : vertex index within one sphere
+        // k : triangle index within one sphere
+        // v0, v1, v2: the three elements of triangle k
+        //
+        // Full vertex buffers contain a total of (kMaxSphereInstanceCount * sphereVertexCount) vertices arranged like so:
+        //
+        // | j(0) | j(1) | ... | j(sphereVertexCount-1) | j(0) | j(1) | ... | j(sphereVertexCount-1) | ... | j(0) | j(1) | ...  | j(sphereVertexCount-1) |
+        // |--------------------i(0)--------------------|--------------------i(1)--------------------| ... |----------i(kMaxSphereInstanceCount)---------|
+        //
+        // Full index buffers contain a total of (kMaxSphereInstanceCount * sphereTriCount * 3) indices arranged like so:
+        //
+        // | v0 | v1 | v2 | v0 | v1 | v2 |    ...     | v0 | v1 | v2 | ... | v0 | v1 | v2 | v0 | v1 | v2 |    ...     | v0 | v1 | v2 |
+        // |     k(0)     |     k(1)     | ... | k(sphereTriCount-1) | ... |     k(0)     |     k(1)     | ... | j(sphereTriCount-1) |
+        // |---------------------------i(0)--------------------------| ... |--------------i(kMaxSphereInstanceCount-1)---------------|
+        //
+        for (uint32_t i = 0; i < kMaxSphereInstanceCount; i++) {
+            uint32_t index = sphereIndices[i];
+            uint32_t x     = (index % (grid.xSize * grid.ySize)) / grid.ySize;
+            uint32_t y     = index % grid.ySize;
+            uint32_t z     = index / (grid.xSize * grid.ySize);
+
+            // Model matrix to be applied to the sphere mesh
+            float4x4 modelMatrix = glm::translate(float3(x * grid.step, y * grid.step, z * grid.step));
+
+            size_t firstVertexOfCurrentSphere = i * sphereVertexCount;
+
+            // For each vertex of the translated sphere, overwrite the position data within large vertex buffers
+            for (uint32_t j = 0; j < sphereVertexCount; ++j) {
+                TriMeshVertexData vertexData = {};
+                mesh.GetVertexData(j, &vertexData);
+                vertexData.position = modelMatrix * float4(vertexData.position, 1);
+
+                TriMeshVertexDataCompressed vertexDataCompressed;
+                vertexDataCompressed.position = half3(glm::packHalf1x16(vertexData.position.x), glm::packHalf1x16(vertexData.position.y), glm::packHalf1x16(vertexData.position.z));
+
+                size_t elementIndex = firstVertexOfCurrentSphere + j;
+                OverwritePositionData(lowInterleavedPositionBufferPtr, vertexDataCompressed, elementIndex);
```

Comment:
From what I can see `lowInterleavedPositionBufferPtr` is only used here and defined 50 lines above, perhaps it is define it on top of this line, or just pass `lowPrecisionInterleaved.GetVertexBuffer(0)` or geometry object itself and have OverWritePositionData get the vertex buffer itself.
Otherwise you have to scroll far up to see where this variable is defined

Reply:
Obsolete